CSharp/Library/Microsoft.Bot.Builder/Dialogs/LuisDialog.cs

```diff
@@ -294,11 +311,24 @@ protected virtual Task<string> GetLuisQueryTextAsync(IDialogContext context, IMe
 #pragma warning restore CS0618
         }
 
-        protected virtual async Task LuisActionDialogFinished(IDialogContext context, IAwaitable<LuisResult> item)
-        {
-            var result = await item;
-            var messageActivity = (IMessageActivity)context.Activity;
-            await DispatchToIntentHandler(context, Awaitable.FromItem(messageActivity), BestIntentFrom(result), result);
+        protected virtual async Task LuisActionDialogFinished(IDialogContext context, IAwaitable<LuisResult> item)
+        {
+            var result = await item;
+            var messageActivity = (IMessageActivity)context.Activity;
+            await DispatchToIntentHandler(context, Awaitable.FromItem(messageActivity), BestIntentFrom(result), result);
+        }
+
+
+        private static async Task EmitTraceInfo(IBotContext context, LuisResult luisResult, ILuisOptions luisOptions, ILuisModel luisModel)
+        {
+            var luisTraceInfo = new LuisTraceInfo
+            {
+                LuisResult = luisResult,
+                LuisOptions = luisOptions,
+                LuisModel = luisModel
+            };
+            var activity = Activity.CreateTraceActivityReply(context.Activity as Activity, "LuisDialog", LuisTraceType, luisTraceInfo, LuisTraceLabel) as IMessageActivity;
```

Comment:
>IMessageActivity [](start = 154, length = 16)

wow, that stoopid :)

Reply:
Agreed, v3 uses IMessageActivities every where and not IActivity, and is very "reply" focused as I told you earlier today :( Can you think of a better way of doing this?